src/extension/src/file_handlers/ExtConfigSettingsHandler.py

```
@@ -33,37 +33,70 @@ def __init__(self, logger, json_file_handler, config_folder):
-            if seq_no is not None:
+            # get seq no from env var
+            seq_no = self.__get_seq_no_from_env_var()
+            if seq_no == -1:
+                self.logger.log_error("Error occurred while verifying sequence number between environment variable and configuration settings")
+                return None
+            elif seq_no is None:
+                if is_enable_request:
+                    self.logger.log_warning("Since sequence number is not provided in the environment variable, will try to fetch from the most recent settings file")
+                else:
+                    self.logger.log_error("Sequence number not provided.")
+                    return None
+            else:
                 return seq_no
 
-            seq_no = None
-            cur_seq_no = None
-            freshest_time = None
-            for subdir, dirs, files in os.walk(self.config_folder):
-                for file in files:
-                    try:
-                        if re.match('^\d+' + self.file_ext + '$', file):
-                            cur_seq_no = int(os.path.basename(file).split('.')[0])
-                            if freshest_time is None:
-                                freshest_time = os.path.getmtime(os.path.join(self.config_folder, file))
-                                seq_no = cur_seq_no
-                            else:
-                                current_file_m_time = os.path.getmtime(os.path.join(self.config_folder, file))
-                                if current_file_m_time > freshest_time:
-                                    freshest_time = current_file_m_time
-                                    seq_no = cur_seq_no
-                    except ValueError:
-                        continue
-            return seq_no
+            # get seq no from settings file
+            return self.__get_seq_no_from_config_settings()
+
         except Exception as error:
```

Why is the actual exception being swallowed?